kernel/src/client/arrow_data.rs

```diff
@@ -152,7 +152,7 @@ impl EngineMap for MapArray {
     }
 }
 
-impl SimpleData {
+impl ArrowEngineData {
     pub fn try_create_from_json(schema: SchemaRef, location: Url) -> DeltaResult<Self> {
         let arrow_schema: ArrowSchema = (&*schema).try_into()?;
```

Comment:
Not sure why this fails to compile on the [ubuntu build](https://github.com/delta-incubator/delta-kernel-rs/actions/runs/8303585598/job/22727948575?pr=145)??
```
error[E0277]: `?` couldn't convert the error to `error::Error`
   --> kernel/src/client/arrow_data.rs:157:62
    |
157 |         let arrow_schema: ArrowSchema = (&*schema).try_into()?;
    |                                                              ^ the trait `std::convert::From<arrow_schema::ArrowError>` is not implemented for `error::Error`
    |
     = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
    = help: the following other types implement trait `std::convert::From<T>`:
              <error::Error as std::convert::From<serde_json::Error>>
              <error::Error as std::convert::From<url::ParseError>>
              <error::Error as std::convert::From<std::io::Error>>
              <error::Error as std::convert::From<parquet::errors::ParquetError>>
    = note: required for `std::result::Result<client::arrow_data::ArrowEngineData, error::Error>` to implement `std::ops::FromResidual<std::result::Result<std::convert::Infallible, arrow_schema::ArrowError>>`
```

Reply:
Ah, I bet it's because the packaging change messed up the [Error](https://github.com/delta-incubator/delta-kernel-rs/blob/main/kernel/src/error.rs#L6-L9) declaration for it:
```
pub enum Error {
    #[cfg(any(feature = "default-client", feature = "simple-client"))]
    #[error("Arrow error: {0}")]
    Arrow(#[from] arrow_schema::ArrowError),
```
s/simple-client/sync-client/ there should fix it?

Reply:
Fixed now.